tests/test_triton_server.py

```diff
@@ -164,6 +167,21 @@ def test_start_stop_gpus(self):
         self.server.stop()
         self.server_local_mock.assert_server_process_terminate_called()
 
+    def test_exception_docker(self):
```

Comment:
`test_perf_analyzer.test_run` has an example of how exceptions are easily mocked. You may not need a whole extra test. This can be put under `test_triton_server.test_start_stop_gpus` as something like 
```
with self.assertRaisesRegex(
            expected_exception=TritonModelAnalyzerException,
            expected_regex="One of the following port(s) are already allocated:",
            msg="Expected TritonModelAnalyzerException"):
    self.server_docker_mock.raise_exception_on_container_run()
    self.server.start()
self.server_docker_mock.reset()          
```